golem-cli/src/command_handler/component/ifs.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::model::app_ext::InitialComponentFile;
+use anyhow::{anyhow, bail, Context};
+use async_zip::tokio::write::ZipFileWriter;
+use async_zip::{Compression, ZipEntryBuilder};
+use golem_common::model::{
+    ComponentFilePath, ComponentFilePathWithPermissions, ComponentFilePathWithPermissionsList,
+};
+use golem_wasm_rpc_stubgen::log::{log_action, LogColorize, LogIndent};
+use std::collections::{HashSet, VecDeque};
+use std::path::PathBuf;
+use tempfile::TempDir;
+use tokio::fs::File;
+use tokio_stream::wrappers::ReadDirStream;
+use tokio_stream::StreamExt;
+
+#[derive(Debug, Clone)]
+pub struct LoadedFile {
+    content: Vec<u8>,
```

Comment:
You are right, we should store a stream here instead of holding the content in memory.
I'll take care of creating a pr for it when I get around to it.